sw/device/lib/base/hardened_status.h

```diff
@@ -23,29 +23,29 @@ extern "C" {
  * This passes `kHardenedBoolTrue` as the status code argument, for extra bits
  * of redundancy in `HARDENED_TRY` and others.
  */
-#define HARDENED_OK_STATUS OK_STATUS(kHardenedBoolTrue)
+#define HARDENED_OK_STATUS ((status_t){.value = kHardenedBoolTrue})
 
 /**
  * Hardened version of the `TRY` macro from `status.h`.
  *
  * @param expr_ An expression that evaluates to a `status_t`.
  * @return The enclosed OK value.
  */
-#define HARDENED_TRY(expr_)                                            \
-  ({                                                                   \
-    status_t status_ = expr_;                                          \
-    if (!(status_ok(status_) && status_.value == kHardenedBoolTrue)) { \
-      return status_;                                                  \
-    }                                                                  \
-    HARDENED_CHECK_EQ(status_.value, kHardenedBoolTrue);               \
-    status_.value;                                                     \
+#define HARDENED_TRY(expr_)                                 \
+  ({                                                        \
+    status_t status_ = expr_;                               \
+    if (hardened_status_ok(status_) != kHardenedBoolTrue) { \
```

Comment:
`hardened_status_ok` is already hardened, so this looks like hardening it twice? Other call-sites that also use `hardened_status_ok` perform an additional hardened comparison. Maybe we should replace it usages with an unhardened macro.